apps/api/src/app/shared/interceptors/cached.interceptor.ts

```diff
@@ -0,0 +1,91 @@
+import { CacheService } from '../services/cache/cache.service';
+
+const cacheService = new CacheService();
+const USECASE_FUNCTION_NAME = 'execute';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function Cached(storeKey?: string) {
+  return (target: any, key: string, descriptor: any) => {
+    const originalMethod = descriptor.value;
+    const context = getKeyContext(key, target);
+
+    descriptor.value = async function (...args: any[]) {
+      const command = args[0];
+
+      const cacheKey = buildKey(context, command);
```

Comment:
Maybe instead of building the `cacheKey` ourselves, we can allow passing the `storeKey` or the "store getter" function, which will be called with all args and will return the key that will be used by the cache? wdyt?
This will give us more flexibility here, and the `buildKey` will be moved outside and will be used in the specific case.